### PR TITLE
apiserver/etcd: add request and request_error metrics

### PR DESCRIPTION
**Pull request by iyear**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Add request and request_error metrics for apiserver etcd storage.

#### Which issue(s) this PR fixes:

Part of #117167 

#### Does this PR introduce a user-facing change?

```release-note
Apiserver adds two new metrics `etcd_requests_total` and `etcd_request_errors_total` that allow users to monitor requests to etcd storage, split by operation and resource type.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wojtek-t** [comment]
Thank you for the PR.
/ok-to-test
/assign

This LGTM, but I would like to get confirmation from SIG instrumentation POV.
@dgrisonnet - can you please take a look at it?

/assign @dgrisonnet

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go [hunk 1]
```diff
@@ -47,6 +47,22 @@ var (
 		},
 		[]string{"operation", "type"},
 	)
+	etcdRequestCounts = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "etcd_request_total",
```
Comment (dgrisonnet) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go:52]: https://prometheus.io/docs/practices/naming/#metric-names
```suggestion
			Name:           "etcd_requests_total",
```

**Review by dgrisonnet — COMMENTED** [review]
Just one comment from my side, otherwise these new metrics look good to me :+1:

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go (hunk 1 above)
Comment (iyear) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go]: Nice suggestion! Updated.

**Comment by wojtek-t** [comment]
/label tide/merge-method-squash

/lgtm
/approve

Thanks!

**Comment by iyear** [comment]
Thank you all:)

**Review by dgrisonnet — APPROVED** [review]
Awesome, great work @iyear :)

/lgtm
/approve

**Comment by shyamjvs** [comment]
@iyear This change deserves a release note for users to monitor the new metric. Can you add one similar to this - https://github.com/kubernetes/kubernetes/pull/117211#issue-1662771094

**Comment by iyear** [comment]
@shyamjvs Added. Thanks for reply:)

**Pull request merged**